homeassistant/components/automation/__init__.py

```diff
@@ -374,52 +496,73 @@ async def async_trigger(self, run_variables, context=None, skip_condition=False)
 
         This method is a coroutine.
         """
-        if self._variables:
-            try:
-                variables = self._variables.async_render(self.hass, run_variables)
-            except template.TemplateError as err:
-                self._logger.error("Error rendering variables: %s", err)
+        reason = ""
+        if "trigger" in run_variables and "description" in run_variables["trigger"]:
+            reason = f' by {run_variables["trigger"]["description"]}'
+        self._logger.debug("Automation triggered%s", reason)
+
+        trigger = run_variables["trigger"] if "trigger" in run_variables else None
+        with trace_automation(
+            self.hass, self.unique_id, self._raw_config, trigger, context
+        ) as automation_trace:
+            if self._variables:
+                try:
+                    variables = self._variables.async_render(self.hass, run_variables)
+                except template.TemplateError as err:
+                    self._logger.error("Error rendering variables: %s", err)
+                    automation_trace.set_error(err)
+                    return
+            else:
+                variables = run_variables
+            automation_trace.set_variables(variables)
+
+            if (
+                not skip_condition
+                and self._cond_func is not None
+                and not self._cond_func(variables)
+            ):
+                self._logger.debug(
+                    "Conditions not met, aborting automation. Condition summary: %s",
+                    condition_trace_get(),
+                )
+                automation_trace.set_condition_trace(condition_trace_get())
                 return
```

Comment:
Shouldn't we always clear, even if condition wasn't met ?

Reply:
If the condition wasn't met, the script's execution stops and we start with a fresh `condition_trace` `ContextVar` if the automation is triggered again.

Test case `test_get_automation_trace` updated to make sure we don't pollute subsequent traces with previous results if conditions fail.